src/core/eval.ml

```diff
@@ -67,7 +67,9 @@ let snf : (term -> term) -> (term -> term) = fun whnf ->
     | Appl(t,u)   -> mk_Appl(snf t, snf u)
     | Meta(m,ts)  -> mk_Meta(m, Array.map snf ts)
     | Patt(i,n,ts) -> mk_Patt(i,n,Array.map snf ts)
-    | Plac _      -> assert false
+    | Plac _      -> h (* may happen when reducing coercions *)
+        (* TODO: the code for coercions should be updated to replace holes by
+           metavariables before reduction. *)
```

Comment:
TODO element. When/where will it be done?

Reply:
This note is not relevant anymore: rewrite rules must generate holes rather than existential variables. What could be done is to develop an SR-check for coercions, so that implicit arguments may instantiated automatically in coercion rules, like they are in rewrite rules.